tasks/reissueZKKYC.ts

```diff
@@ -0,0 +1,169 @@
+  let newZkKYC = new ZKCertificate(args.holderCommitment, ZkCertStandard.zkKYC, eddsa, args.randomSalt, newZkKYCFields);
+
+  // let provider sign the zkKYC
+  const providerEdDSAKey = await getEddsaKeyFromEthSigner(provider);
+  oldZkKYC.signWithProvider(providerEdDSAKey);
+  newZkKYC.signWithProvider(providerEdDSAKey);
+
+  if (args.registryAddress === undefined) {
+    console.log(chalk.yellow("Parameter 'registry-address' is missing. The zkKYC has not been issued on chain"));
+    return;
+  }
+
+  console.log("reissuing zkKYC...");
+  const recordRegistry = await hre.ethers.getContractAt('KYCRecordRegistry', args.registryAddress) as KYCRecordRegistry;
+  const guardianRegistry = await hre.ethers.getContractAt('KYCCenterRegistry', await recordRegistry._KYCCenterRegistry()) as KYCCenterRegistry;
+
+  if (!(await guardianRegistry.KYCCenters(provider.address))) {
+    throw new Error(`Provider ${provider.address} is not a guardian yet. Please register it first using the script .`);
+  }
+  const oldLeafBytes = fromHexToBytes32(fromDecToHex(oldZkKYC.leafHash));
+  const newLeafBytes = fromHexToBytes32(fromDecToHex(newZkKYC.leafHash));
+
+  if (!args.merkleProof) {
+    console.log("zkKYC", zkKYC.exportJson());
+    console.log(chalk.yellow("Merkle proof generation is disabled. Before using the zkKYC, you need to generate the merkle proof."));
+    return;
+  }
+
+  console.log("Generating merkle proof. This might take a while because it needs to query on-chain data...");
+  // Note for developers: The slow part of building the Merkle tree can be skipped if you build a back-end service maintaining an updated Merkle tree
+  const poseidon = await buildPoseidon();
+  const merkleDepth = 32;
+  const leafLogResults = await queryOnChainLeaves(hre.ethers, recordRegistry.address); // TODO: provide first block to start querying from to speed this up
+  const leafHashes = leafLogResults.map(x => x.leafHash);
+  const leafIndices = leafLogResults.map(x => x.index);
+  const merkleTree = new SparseMerkleTree(merkleDepth, poseidon);
+  const batchSize = 10_000;
+  console.log(`Adding leaves to the merkle tree`);
+  for (let i = 0; i < leafLogResults.length; i += batchSize) {
+    merkleTree.insertLeaves(leafHashes.slice(i, i + batchSize), leafIndices.slice(i, i + batchSize));
+  }
+
+  if (merkleTree.retrieveLeaf(0, args.index) !== oldLeafBytes) {
+    console.log(`the current leaf hash at index ${args.index} does not correpond with the outdated zkKYC Cert we want to update`);
+    return;
+  }
+
+  // create Merkle proof
+  const merkleProof = merkleTree.createProof(index);
+  let output = zkKYC.export();
+  output.merkleProof = {
+    root: merkleTree.root,
+    pathIndices: merkleProof.pathIndices,
+    pathElements: merkleProof.path,
+  }
+    
+  // now we have the merkle proof to add a new leaf
+  let tx = await recordRegistry.addZkKYCRecord(index, newleafBytes, merkleProof.path.map(x => fromHexToBytes32(fromDecToHex(x))));
```

Comment:
There is a typo in `newleafBytes`. The variable has been defined with a capital `L`.

Were you able to test the script? I am happy to help you test it on the devnet.